### PR TITLE
Enhance test utility for running tasks in parallel

### PR DESCRIPTION
**Pull request by original-brownbear**
Follow up to #110552, add utility for starting tasks at the same time via a barrier as discussed there. Also, make use of the new tooling in a couple more spots to save LoC and thread creation.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java [hunk 1]
```diff
@@ -2434,6 +2435,22 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
...
 
+    /**
+     * Same as {@link #runInParallel(int, IntConsumer)} but also attempts to start all tasks at the same time by blocking execution on a
+     * barrier until all threads are started and ready to execute their task.
+     */
+    public static void startInParallel(int numberOfTasks, IntConsumer taskFactory) throws InterruptedException {
+        final CyclicBarrier barrier = new CyclicBarrier(numberOfTasks);
+        runInParallel(numberOfTasks, i -> {
+            try {
+                barrier.await();
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:2446]: I think I'd use a `safeAwait` here to save the try/catch noise and (preserve the interrupt flag properly fwiw).

**Review by DaveCTurner — APPROVED** [review]
LGTM (one nit)

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java (hunk 1 above)
Comment (original-brownbear) [reply test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java]: ++ done

**Comment by original-brownbear** [comment]
Thanks David!

**Comment by original-brownbear** [comment]
Someone else asked the same in the original PR :) I'd rather not use the barrier throughout, it slows down measurably for machines with a very large core count (like my personal workstation :P) so it's nice to only force the synchronisation when we care IMO.

**Pull request merged**